dev/source/docs/building-setup-windows.rst

```diff
@@ -4,20 +4,7 @@
 Setup the Build Environment on Windows
 ======================================
 
-There are two options for building on Windows: Windows Subsystem for Linux (WSL) or Cygwin. New developers should use WSL as it offers faster compilation times, greater compatibility, and a larger support base from other developers.
+Cygwin and Eclipse are no longer recommended for building code on a Windows PC. WSL2 is now the preferred environment.
```

Comment:
Cygwin must be supported or SITL in Mission Planner would no longer work. The sentence that is already there covers what is in practice.

Reply:
the general consensus so far has been to have the wiki direct new devs to use WSL in Windows for builds....and VSCode for an IDE in Windows...and to sim with SITL in WSL in Windows in conjunction multiple graphical/physics interfaces like RealFlight...non-devs are the ones usually asking to use MP sims, so they dont really need to build an ELF.... but info on that using github actions is in the wiki, although needs to be made clearer

I think we should structure the wiki to point down the above  path for Windows boxes...but not deprecate the CYGWIN/ECLIPSE setup and CYGWIN build but put those pages for advanced users...

We need some new info, which I think @rmackay9 is working on:
- setup VSCODE with links to WSL in windows for an IDE
- how to use the VSCODE deug hooks in that environment
also how to setup WSL2 in Win11 (which I have already done, but did note the steps....will have to create a VM and write down the steps for a wiki page)

We get very few support questions about setting up a Linux box, but do get them from Windows users...at least my impression over the last few years.....especially having issues with Cygwin builds...haven't kept track since when those arise someone points them to WSL

Reply:
This pages gives decently detailed debug info.  on the use cases I've ran across. https://ardupilot.org/dev/docs/debugging-with-gdb-using-vscode.html

I would still be tempted to deprecate Eclipse, if 0 devs are actually using it? (If we get Randy to convert ;) )